radix-engine/src/engine/heap.rs

```diff
@@ -123,6 +123,10 @@ impl Heap {
             .remove(&node_id)
             .ok_or(CallFrameError::RENodeNotOwned(node_id))
     }
+
+    pub fn nodes(&self) -> &HashMap<RENodeId, HeapRENode> {
+        &self.nodes
+    }
```

Comment:
Instead of asking heap for global node list, we should check the owned the nodes by the current frame.

Reply:
_(for the record: the tracing architecture has been changed, as discussed; this method no longer exists)_